test-snapshots/snapshots-output/Snapshot.ConstructorAccessor.ss

```diff
@@ -0,0 +1,97 @@
+#!r6rs
+#!chezscheme
+(library
+  (Snapshot.ConstructorAccessor lib)
+  (export
+    First*
+    First-value0
+    First?
+    Foo
+    Foo*
+    Foo-value0
+    Foo-value1
+    Foo-value2
+    Foo?
+    HasArgs
+    HasArgs*
+    HasArgs-value0
+    HasArgs-value1
+    HasArgs-value2
+    HasArgs?
+    Last*
+    Last-value0
+    Last?
+    NoArgs
+    NoArgs?
+    test1
+    test2
+    test3
+    test4
+    test5
+    x
+    y
+    z)
+  (import
+    (prefix (chezscheme) scm:)
+    (prefix (_Chez_Runtime lib) rt:))
+
+  (scm:define-record-type (First$ First* First?)
+    (scm:fields (scm:immutable value0 First-value0)))
+
+  (scm:define-record-type (Last$ Last* Last?)
+    (scm:fields (scm:immutable value0 Last-value0)))
+
+  (scm:define NoArgs
+    (scm:quote NoArgs))
+
+  (scm:define NoArgs?
+    (scm:lambda (v)
+      (scm:eq? (scm:quote NoArgs) v)))
+
+  (scm:define-record-type (HasArgs$ HasArgs* HasArgs?)
+    (scm:fields (scm:immutable value0 HasArgs-value0) (scm:immutable value1 HasArgs-value1) (scm:immutable value2 HasArgs-value2)))
+
+  (scm:define HasArgs
+    (scm:lambda (value0)
+      (scm:lambda (value1)
+        (scm:lambda (value2)
+          (HasArgs* value0 value1 value2)))))
+
+  (scm:define-record-type (Foo$ Foo* Foo?)
+    (scm:fields (scm:immutable value0 Foo-value0) (scm:immutable value1 Foo-value1) (scm:immutable value2 Foo-value2)))
+
+  (scm:define Foo
+    (scm:lambda (value0)
+      (scm:lambda (value1)
+        (scm:lambda (value2)
+          (Foo* value0 value1 value2)))))
+
+  (scm:define z
+    (Foo* 1 1 1))
```

Comment:
Just highlighting here that the uncurried function is used when all args are provided to the constructor. :heavy_check_mark: